src/main/java/ch/njol/skript/expressions/ExprResult.java

```diff
@@ -0,0 +1,115 @@
+		"If the thing is expected to return multiple values, use 'results' instead of 'result'."
+})
+@Examples({
+		"set {_function} to the function named \"myFunction\"",
+		"set {_result} to the result of {_function}",
+		"set {_list::*} to the results of {_function}",
+		"set {_result} to the result of {_function} with arguments 13 and true"
+})
+@Since("INSERT VERSION")
+@Keywords({"run", "result", "execute", "function", "reflection"})
+public class ExprResult extends PropertyExpression<Executable<Event, Object>, Object> {
+
+	static {
+		Skript.registerExpression(ExprResult.class, Object.class, ExpressionType.COMBINED,
+			"[the] result[plural:s] of [running|executing] %executable% [arguments:with arg[ument]s %-objects%]");
+	}
+
+	private Expression<?> arguments;
+	private boolean hasArguments, isPlural;
+	private DynamicFunctionReference.Input input;
+
+	@Override
+	public boolean init(Expression<?>[] expressions, int matchedPattern, Kleenean isDelayed,
+						ParseResult result) {
+		if (!this.getParser().hasExperiment(Feature.SCRIPT_REFLECTION))
+			return false;
+		//noinspection unchecked
+		this.setExpr((Expression<? extends Executable<Event, Object>>) expressions[0]);
+		this.hasArguments = result.hasTag("arguments");
+		this.isPlural = result.hasTag("plural");
+		if (hasArguments) {
+			this.arguments = LiteralUtils.defendExpression(expressions[1]);
+			Expression<?>[] arguments;
+			if (this.arguments instanceof ExpressionList<?> list) {
+				arguments = list.getExpressions();
+			} else {
+				arguments = new Expression[]{this.arguments};
+			}
+			this.input = new DynamicFunctionReference.Input(arguments);
+			return LiteralUtils.canInitSafely(this.arguments);
+		} else {
+			this.input = new DynamicFunctionReference.Input();
+		}
+		return true;
+	}
+
+	@Override
+	protected Object[] get(Event event, Executable<Event, Object>[] source) {
+		for (Executable<Event, Object> task : source) {
+			Object[] arguments;
+			//noinspection rawtypes
+			if (task instanceof DynamicFunctionReference reference) {
+				Expression<?> validated = reference.validate(input);
+				if (validated == null)
+					return new Object[0];
+				arguments = validated.getArray(event);
+			} else if (hasArguments) {
+				arguments = this.arguments.getArray(event);
+			} else {
+				arguments = new Object[0];
+			}
+			Object execute = task.execute(event, arguments);
+			if (execute instanceof Object[] results)
+				return results;
+			return new Object[]{execute};
+		}
+		return new Object[0];
+	}
+
+	@Override
+
+	public Class<?> @Nullable [] acceptChange(Changer.ChangeMode mode) {
+		return null;
+	}
+
+	@Override
+	public Class<Object> getReturnType() {
+		return Object.class;
+	}
+
+	@Override
+	public boolean isSingle() {
+		return !isPlural;
+	}
+
+	@Override
+	public String toString(@Nullable Event event, final boolean debug) {
+		String text = "the result" + (isPlural ? "s" : "") + " of " + getExpr().toString(event, debug);
```

Comment:
SyntaxStringBuilder

Reply:
In its current state I don't think SyntaxStringBuilder has any use apart from getting to skip the `toString` bits, since you can't chain append calls.
<img width="244" alt="image" src="https://github.com/user-attachments/assets/457e76ad-f9ea-44ca-8331-610259ad1c2b" />

It's also not useful with propertyexpressions right now because of the annotation mismatch.
<img width="218" alt="image" src="https://github.com/user-attachments/assets/2810755c-1c1a-4879-998b-e5392545fc11" />

Right now, it's just significantly more mess than doing it normally.
<img width="568" alt="image" src="https://github.com/user-attachments/assets/b76df5f1-4c6c-401d-b703-f27997ae9c5e" />
<img width="764" alt="image" src="https://github.com/user-attachments/assets/cb96c8c0-bd8a-41c5-8cee-cf61f3cda832" />
Once these shortcomings have been improved I might consider using it.

Reply:
Most of those issues have been fixed, so you should be fine to use it now. Rather than chaining (although now supported), you can just pass multiple arguments:
```java
builder.append("with arguments", arguments);
```